pd/src/main.rs

```diff
@@ -306,16 +306,22 @@ async fn main() -> anyhow::Result<()> {
                                     name: v.name.clone(),
                                     website: v.website.clone(),
                                     description: v.description.clone(),
-                                    funding_streams: v
-                                        .funding_streams
-                                        .iter()
-                                        .map(|fs| FundingStream {
+                                    funding_streams: FundingStreams::try_from(
+                                        v.funding_streams
+                                            .iter()
+                                            .map(|fs| {
+                                                FundingStream {
                                             address: Address::from_str(&fs.address).expect(
                                                 "invalid funding stream address in validators.json",
                                             ),
                                             rate_bps: fs.rate_bps,
-                                        })
-                                        .collect(),
+                                        }
+                                            })
+                                            .collect::<Vec<FundingStream>>(),
+                                    )
+                                    .expect(
```

Comment:
Likewise here, it would be better (in general) to use `anyhow` to throw errors rather than `expect`.